### PR TITLE
Added example in map-reduce markdown and samples directory - Issue 156

### PR DESCRIPTION
**Pull request by mdolz**
Added an example in map_reduce for the canonical word frequency example.

### CODE REVIEW

File: samples/map-reduce/word_count/main.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <chrono>
+#include <string>
+#include <numeric>
+#include <stdexcept>
+
+// grppi
+#include "poly/polymorphic_execution.h"
```
Comment (jdgarciauc3m) [inline samples/map-reduce/word_count/main.cpp]: Not needed: Remove.

File: samples/map-reduce/word_count/main.cpp [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+#include "mapreduce.h"
+#include "poly/map.h"
```
Comment (jdgarciauc3m) [inline samples/map-reduce/word_count/main.cpp]: Not needed: Remove.

File: samples/map-reduce/word_count/main.cpp (hunk 1 above)
Comment (mdolz) [reply samples/map-reduce/word_count/main.cpp]: Removed.

File: samples/map-reduce/word_count/main.cpp (hunk 2 above)
Comment (mdolz) [reply samples/map-reduce/word_count/main.cpp]: Removed

**Review by jdgarciauc3m — APPROVED** [review]
OK

**Comment by jdgarciauc3m** [comment]
Solves issue #156

**Pull request merged**